cpp/src/strings/utilities.cu

```diff
@@ -163,16 +164,26 @@ struct character_flags_singleton
 private:
 
     character_flags_table_type* d_character_flags_table{};
+    bool b_rmm_allocated{false};
 
     character_flags_singleton()
     {
-        RMM_TRY(RMM_ALLOC(&d_character_flags_table,sizeof(g_character_codepoint_flags),0));
+        rmmOptions_t options;
+        if( rmmIsInitialized(&options) && (options.allocation_mode == PoolAllocation) )
```

Comment:
I think making this a singleton is a problem. Singletons often bring along problems, so let's avoid it if possible.

Specifically, if RMM is initialized to pool mode, then this singleton is created, and then RMM is reinitialized (to any mode), this singleton's memory will be freed out from under it.

Instead, let's just mark the memory as __device__.

Reply:
Replaced singleton with `__device__` memory declaration.